server/matchmaker/search.py

```diff
@@ -234,10 +235,20 @@ def cancel(self):
         """
         self._match.cancel()
 
-    def __str__(self):
-        return "Search({}, {}, {})".format(self.players, self.match_threshold, self.search_expansion)
+    def __str__(self) -> str:
+        return (
+            f"Search({self.rating_type}, {self._players_repr()}, threshold="
+            f"{self.match_threshold:.2}, expansion={self.search_expansion:.2})"
+        )
+
+    def _players_repr(self) -> str:
+        contents = ', '.join(
+            f"Player({p.login}, {p.id}, {p.ratings[self.rating_type]})"
```

Comment:
`p.ratings[self.rating_type]` will be the un-adjusted rating for newbie players, so not necessarily the one that is actually used for matching in `self.ratings`. If the rating actually used is more useful here then maybe it would be nice to refactor `self.adjusted_rating` to do the right thing (TM) for both newbie players and experienced players and then you could use `self.adjusted_rating(player)` here. Then `self.ratings` can also just use the `adjusted_rating` for all players.
(Should hopefully be a change that stays internal to `search.py`)

Reply:
I think using the actual rating is better... I'm not entirely sure, but the adjusted rating is also logged separately so this way there is more info (both adjusted and non adjusted rating is available)

Reply:
True, although the adjusted rating might only be logged if there's a match